src/test/java/randomgen/json/RandomJsonWhiteSpaceInjector.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    @Nonnull
+    private static byte[] insertRandomWhiteSpaces(List<Integer> indexesToInsertWhiteSpaces, byte[] originalJsonBytes) {
+        final byte[] resultingJsonBytes = new byte[indexesToInsertWhiteSpaces.size() + originalJsonBytes.length];
+        indexesToInsertWhiteSpaces.sort((a, b) -> Integer.compare(a, b));
+        int indexInOriginalByteArray = 0;
+        int indexInResultByteArray = 0;
+        int whiteSpacesInjectedSoFar = 0;
+        while (!indexesToInsertWhiteSpaces.isEmpty()) {
+            int indexToInsert = indexesToInsertWhiteSpaces.get(0);
+            System.arraycopy(
+                    originalJsonBytes,
+                    indexInOriginalByteArray,
+                    resultingJsonBytes,
+                    indexInResultByteArray,
+                    indexToInsert - indexInOriginalByteArray);
+            resultingJsonBytes[indexToInsert + whiteSpacesInjectedSoFar] = getRandomWhiteSpaceByte();
+            whiteSpacesInjectedSoFar++;
+            indexInOriginalByteArray = indexToInsert;
+            indexInResultByteArray = indexToInsert + whiteSpacesInjectedSoFar;
+            indexesToInsertWhiteSpaces.remove(0);
+        }
+        System.arraycopy(
+                originalJsonBytes,
+                indexInOriginalByteArray,
+                resultingJsonBytes,
+                indexInResultByteArray,
+                originalJsonBytes.length - indexInOriginalByteArray);
+        return resultingJsonBytes;
+    }
+
+    private static byte getRandomWhiteSpaceByte() {
+        int randomInt = new Random().nextInt(4);
+        return switch (randomInt) {
+            case 0 -> CARRIAGE_RETURN.getAsciiByteValue();
+            case 1 -> HORIZONTAL_TAB.getAsciiByteValue();
+            case 2 -> LINE_FEED.getAsciiByteValue();
+            case 3 -> SPACE.getAsciiByteValue();
+            default -> throw new IllegalStateException("Unexpected value to get random whitespace byte: " + randomInt);
+        };
+    }
+
+    /**
+     * Checks whether a whitespace can injected before the provided index in the provided JsonBytes.
+     * To keep it simple, this method is sometimes more strict than it needs to be (it returns false
+     * while it was possible, for example in Strings). However, it will never return true if that
+     * would result in invalid JSON if a whitespace would be injected before the provided index.
+     *
+     * @param jsonBytes the JSON bytes for which it will be checked if the whitespace can be
+     *     injected
+     * @param index the index for which it will be checked, must be > 1
+     * @return whether injecting a whitespace before the index in the provided JSON bytes will
+     *     result in valid JSON
+     */
+    private static boolean canInjectWhiteSpaceBeforeByteAtIndex(byte[] jsonBytes, int index) {
+        if (index < 1) {
+            throw new IllegalArgumentException(
```

Comment:
Can't you inject a whitespace before object start?

Reply:
You can, but honestly I don't really care about it. I just wanted to make sure there will always be a previous byte (I was only interested in whitespaces inside the JSON)

Reply:
Actually, I updated it so it is possible now.